### ISSUE TITLE
Strikethrough support in Markdown

### ISSUE THREAD

**Issue by oblitum** [valentjn/languagetool-languageserver#123]
**Describe the bug**

**Steps to reproduce**

1. Write any strikethrough text, as in ~~strikethrough~~ (`~~strikethrough~~`).

**Expected behavior**

Even though not present in the original Markdown and CommonMark, it's present in many well-known variants like GitHub Flavored Markdown, etc. It would be nice to have support for it instead of it being picked as a spelling mistake.

**Sample document**
If the bug occurs for a specific document (e.g. LaTeX), please paste it here. If your document is very long or confidential, please create and attach a smaller example for which the bug still occurs so that we can reproduce it.

<details>

~~~
# markmap

## Links

- <https://markmap.js.org/>
- [GitHub](https://github.com/gera2ld/markmap)

## Related

- [coc-markmap](https://github.com/gera2ld/coc-markmap)
- [gatsby-remark-markmap](https://github.com/gera2ld/gatsby-remark-markmap)

## Features

- links
- **inline** ~~text~~ *styles*
- multiline
  text
- `inline code`
-
  ```js
  console.log('code block');
  ```
- KaTeX - $x = {-b \pm \sqrt{b^2-4ac} \over 2a}$
~~~

</details>

**LTeX configuration**
Please paste all configuration settings starting with `ltex.` from your `settings.json`. You can help us by temporarily removing some irrelevant settings from your `settings.json` and see if the bug still occurs.

<details>

```
```

</details>

**LTeX LS log**
Paste the log (output) of LTeX LS here:

<details>

```
[Info  - 9:04:54 PM] Starting ltex-ls...
Nov 14, 2021 9:04:56 PM org.bsplines.ltexls.server.LtexLanguageServer initialize
INFO: ltex-ls 15.0.0 - initializing...
Nov 14, 2021 9:04:56 PM org.bsplines.ltexls.tools.I18n setLocale
INFO: Setting locale to ''
Nov 14, 2021 9:04:56 PM org.bsplines.ltexls.settings.SettingsManager$Companion logDifferentSettings
FINE: Reinitializing LanguageTool due to different settings for language 'en-US': setting 'additionalRules.languageModel', old '', new '/home/francisco/Downloads/ngrams'
Nov 14, 2021 9:04:57 PM org.bsplines.ltexls.server.DocumentChecker logTextToBeChecked
FINE: Checking the following text in language 'en-US' via LanguageTool: "markmap\n\nLinks\n\nDummy0\nGitHub\n\nRelated\n\ncoc-markmap\ngatsby-remark-markmap\n\nFeatures\n\nlinks\ninline ~~"... (truncated to 100 characters)
Nov 14, 2021 9:04:58 PM org.bsplines.ltexls.server.DocumentChecker checkAnnotatedTextFragment
FINE: Obtained 5 rule matches
Nov 14, 2021 9:10:21 PM org.bsplines.ltexls.server.DocumentChecker logTextToBeChecked
FINE: Checking the following text in language 'en-US' via LanguageTool: "markmap\n\nLinks\n\nDummy0\nGitHub\n\nRelated\n\ncoc-markmap\ngatsby-remark-markmap\n\nFeatures\n\nlinks\ninline ~~"... (truncated to 100 characters)
Nov 14, 2021 9:10:21 PM org.bsplines.ltexls.server.DocumentChecker checkAnnotatedTextFragment
FINE: Obtained 5 rule matches
Nov 14, 2021 9:10:23 PM org.bsplines.ltexls.server.DocumentChecker logTextToBeChecked
FINE: Checking the following text in language 'en-US' via LanguageTool: "markmap\n\nLinks\n\nDummy0\nGitHub\n\nRelated\n\ncoc-markmap\ngatsby-remark-markmap\n\nFeatures\n\nlinks\ninline ~~"... (truncated to 100 characters)
Nov 14, 2021 9:10:23 PM org.bsplines.ltexls.server.DocumentChecker checkAnnotatedTextFragment
FINE: Obtained 5 rule matches
```

</details>

**Version information**
List here the version information of the relevant software.

- Operating system: Arch Linux
- ltex-ls: 15.0.0
- Java: "Using Java bundled with ltex-ls as ltex.java.path is not set."

**Additional context/information**
You can add any other context or information about the problem here.

![gif-2021-11-15-183231](https://user-images.githubusercontent.com/1269815/141856899-5bf4bc78-5ecd-4792-ae3d-db7cb6e25b9f.gif)

**Comment by oblitum** [comment]
I was just not sure to classify this as bug or feature-request.

**Comment by valentjn** [comment]
Classifying as feature request as the docs don't say that GFM is supported.

**Comment by valentjn** [comment]
Fix released in 15.2.0.
